ledger/acctonline.go

```diff
@@ -118,13 +118,19 @@ type onlineAccounts struct {
 
 	// disableCache (de)activates the LRU cache use in onlineAccounts
 	disableCache bool
+
+	// cache for expired online circulation stake since the underlying query is quite heavy
+	expiredCirculationCache *expiredCirculationCache
 }
 
 // initialize initializes the accountUpdates structure
 func (ao *onlineAccounts) initialize(cfg config.Local) {
 	ao.accountsReadCond = sync.NewCond(ao.accountsMu.RLocker())
 	ao.acctLookback = cfg.MaxAcctLookback
 	ao.disableCache = cfg.DisableLedgerLRUCache
+	// 2 pages * 256 entries look large enough to handle
+	// both early and late votes, and well as a current and previous stateproof periods
+	ao.expiredCirculationCache = makeExpiredCirculationCache(256)
```

Comment:
Do items in this cache ever get removed?

Reply:
yes, cur/prev pages alternate in `expiredCirculationCache.put`